zenoh-flow/src/model/record/dataflow.rs

```diff
@@ -248,19 +248,18 @@ impl DataFlowRecord {
                 // We only create a sender if none was created for the same resource. The rationale
                 // is to avoid creating multiple publisher for the same resource in case an operator
                 // acts as a multiplexor.
-                if !self
-                    .connectors
-                    .iter()
-                    .any(|c| c.kind == ZFConnectorKind::Sender && c.resource == z_resource_name)
-                {
+                if !self.connectors.iter().any(|(_id, c)| {
+                    c.kind == ZFConnectorKind::Sender && c.resource == z_resource_name
+                }) {
                     // creating sender
                     let sender_id = format!(
                         "sender-{}-{}-{}-{}",
                         &self.flow, &self.uuid, &l.from.node, &l.from.output
-                    );
+                    )
+                    .into();
                     let sender = ZFConnectorRecord {
                         kind: ZFConnectorKind::Sender,
-                        id: sender_id.clone().into(),
+                        id: Arc::clone(&sender_id),
```

Comment:
I prefer to have an explicit call to `Arc::clone` as we know that it is just a reference increment and not an expensive operation.
I’ll revert this changes as, as you pointed out, it can be hidden in the "newtype" we intend on creating.